hnn_core/network.py

```diff
@@ -764,6 +793,254 @@ def _get_src_type_and_pos(self, gid):
+            allow_autapses=False)
+
+        src_cell = 'L5Pyr'
+        nc_dict['lamtha'] = 3.
+        key = f'gbar_L5Pyr_{target_cell}'
+        nc_dict['A_weight'] = self.params[key]
+        loc = 'soma'
+        receptor = 'ampa'
+        self._all_to_all_connect(
+            connectivity, src_cell, target_cell, loc, receptor, nc_dict)
+
+        src_cell = 'L2Pyr'
+        key = f'gbar_L2Pyr_{target_cell}'
+        nc_dict['A_weight'] = self.params[key]
+        loc = 'soma'
+        receptor = 'ampa'
+        self._all_to_all_connect(
+            connectivity, src_cell, target_cell, loc, receptor, nc_dict)
+
+    def _all_to_all_connect(self, connectivity_list, src_cell, target_cell,
+                            loc, receptor, nc_dict,
+                            allow_autapses=True, unique=False):
+        """Generate connectivity list given lists of sources and targets.
+
+        Parameters
+        ----------
+        connectivity_list : list of dict
+            The connectivity list to be updated
+        src_cell : str
+            Source cell type.
+        target_cell : str
+            Target cell type.
+        loc : str
+            If 'proximal' or 'distal', the corresponding
+            dendritic sections from Cell.sect_loc['proximal']
+            or Cell.Sect_loc['distal'] are used
+        receptor : str
+            The receptor.
+        nc_dict : dict
+            The connection dictionary containing keys
+            A_delay, A_weight, lamtha, and threshold.
+        allow_autapses : bool
+            If True, allow connecting neuron to itself.
+        unique : bool
+            If True, each target cell gets one "unique" feed.
+            If False, all src_type cells are connected to
+            all target_type cells.
+        """
+        connectivity = list()
+
+        src_gids = self.gid_ranges[_long_name(src_cell)]
+        target_gids = self.gid_ranges[_long_name(target_cell)]
+        src_types = np.repeat(src_cell, len(src_gids))
+        target_types = np.repeat(target_cell, len(target_gids))
+
+        src_start = src_gids[0]  # Necessary for unique feeds
+        for target_type, target_gid in zip(target_types, target_gids):
+            if unique:
+                src_gids = [target_gid + src_start]
+                src_types = [src_types[0]]  # Assumes that all src_types match
+            for src_type, src_gid in zip(src_types, src_gids):
+                conn = {'src_type': src_type, 'src_gid': src_gid,
+                        'target_type': target_type, 'target_gid': target_gid,
+                        'loc': loc, 'receptor': receptor,
+                        'nc_dict': deepcopy(nc_dict)}
+                if not allow_autapses and src_gid == target_gid:
+                    continue
+
+                connectivity.append(conn)
+        connectivity_list.extend(connectivity)
+
+    def add_connection(self, src_gid, target_gid, loc, receptor, delay,
```

Comment:
It looks like it might be possible to add a connection on top of a previously existing connection here. If this is true, could we add a quick assertion to this function that throws an error message if the user tries adding a connection between two cell structures that already exists?

Reply:
I see the logic, but I don't think this is necessary. It's totally plausible that two neurons are connected via multiple branches from the same axon. More importantly, it is necessary to have multiple connections to achieve synapses with multiple ion channels.

Reply:
> It's totally plausible that two neurons are connected via multiple branches from the same axon

but this can be represented as a higher weight?

> More importantly, it is necessary to have multiple connections to achieve synapses with multiple ion channels.

right but in `network_builder.ncs` it gets a [unique key](https://github.com/jonescompneurolab/hnn-core/blob/master/hnn_core/network_builder.py#L446) as far as I remember. I think we should reduce surprises as much as possible rather than cater to 5% of usecases.

Reply:
> but this can be represented as a higher weight?
Hmm not if there are different delays. I'll ponder this but personally it seems more surprising to force a single connection between neurons. 

Also implementing this check would require a pretty big overhaul of the connectivity storage system as you would need a dict of dicts to avoid looping over the entire list every time `add_connection()` gets called. If everyone feels strongly about this perhaps a followup PR converting `net.connectivity` to a dict structure?